lib/cartopy/img_transform.py

```diff
@@ -268,17 +268,25 @@ def regrid(array, source_x_coords, source_y_coords, source_proj, target_proj,
                                               target_x_points.flatten(),
                                               target_y_points.flatten())
 
+    # Find mask of valid points before querying kdtree: scipy >= 1.11 errors
+    # when querying nan points, might as well use for pykdtree too.
+    indices = np.zeros(target_xyz.shape[0], dtype=int)
+    finite_xyz = np.all(np.isfinite(target_xyz), axis=-1)
+
     if _is_pykdtree:
-        kdtree = pykdtree.kdtree.KDTree(xyz)
+        kdtree = _kdtree_handle(xyz)
         # Use sqr_dists=True because we don't care about distances,
         # and it saves a sqrt.
-        _, indices = kdtree.query(target_xyz, k=1, sqr_dists=True)
+        _, indices[finite_xyz] = kdtree.query(target_xyz[finite_xyz, :],
+                                              k=1,
+                                              sqr_dists=True)
```

Comment:
Since the object creation and query are so similar, do you think it'd be worth just setting the different keyword arguments in a dict and then unpacking for the same call function below. Something like this:

```python
if _is_pykdtree:
    class_kws = {}
    query_kws = {'k': 1, 'sqr_dists': True}
else:
    class_kws = {'balanced_tree': False}
    query_kws = {'k': 1}

kdtree = _kdtree_handle(xyz, **class_kws)
kdtree.query(target_xyz, **query_kws)
```

I think it is fine as-is too, just thought I'd throw that other option out there as well.

Reply:
I don't have a strong feeling on it. Both versions seem clear enough to me, happy to defer to you on this.

Reply:
I'd suggest just leaving it as you have it currently then and ignore this comment.